django/db/models/fields/json.py

```diff
@@ -385,10 +385,16 @@ def as_sqlite(self, compiler, connection):
         datatype_values = ",".join(
             [repr(datatype) for datatype in connection.ops.jsonfield_datatype_values]
         )
-        return (
-            "(CASE WHEN JSON_TYPE(%s, %%s) IN (%s) "
-            "THEN JSON_TYPE(%s, %%s) ELSE JSON_EXTRACT(%s, %%s) END)"
-        ) % (lhs, datatype_values, lhs, lhs), (tuple(params) + (json_path,)) * 3
+        if connection.get_database_version() >= (3, 38):
```

Comment:
I'd change it to the SQLite-specific feature flag.

Reply:
I'll add the feature flag if we decide to move forward with this (https://github.com/django/django/pull/18241#discussion_r1687995539)